benches/benches/vm.rs

```diff
@@ -33,14 +32,12 @@ where
                 instruction,
                 diff,
             } = &mut i;
-            let original_db = vm.as_mut().database_mut().clone();
-            let mut db_txn = {
-                let db = vm.as_mut().database_mut();
-                let db_txn = db.transaction();
-                // update vm database in-place to use transaction
-                *db = db_txn.as_ref().clone();
-                db_txn
-            };
+            let checkpoint = vm
+                .as_mut()
+                .database_mut()
+                .checkpoint()
+                .expect("Should be able to create a checkpoint");
+            let original_db = core::mem::replace(vm.as_mut().database_mut(), checkpoint);
```

Comment:
Is this really representative of real world conditions though? Since we'll pretty much always be executing this opcode from within a db transaction when going through the block executor.

Reply:
<img width="718" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/de162620-3b3c-474a-a519-793e559bf501">

The idea is to avoid optimizations from `MemoryTransactionView` in the case of non-first interaction with the storage key. Using checkpoint instead of  `MemoryTransactionView`, we go to `self.data_source.get(key, column)` part directly.

But yeah, it is not real-world execution. Because inside of the block, we have `RocksDB -> MemoryTransactionView -> MemoryTransactionView`. We can create a special `BenchmarkMemoryTransactionView` for benchmarks that does all the same thing, except it always goes to `data_source`. But I think the overhead from `MemoryTransactionView` is much lesser than direct access to the checkpoint database. The proof is a significant increase in the execution time of the opcode for case with `1`.

<img width="673" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/cef3fb4d-339e-49b6-a9aa-4168a092c6b2">